model-integration/src/main/java/ai/vespa/embedding/support/ModelPathHelperImpl.java

```diff
@@ -0,0 +1,78 @@
+package ai.vespa.embedding.support;
+
+import ai.vespa.secret.Secret;
+import ai.vespa.secret.Secrets;
+import com.yahoo.component.annotation.Inject;
+import com.yahoo.config.ModelReference;
+import com.yahoo.config.UrlReference;
+import com.yahoo.vespa.config.UrlDownloader;
+import com.yahoo.vespa.config.UrlDownloader.DownloadOptions;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+
+public class ModelPathHelperImpl implements ModelPathHelper {
+
+    public static final Duration MODEL_DOWNLOAD_TIMEOUT = Duration.ofMinutes(60);
+
+    private final Secrets secrets;
+    private final ModelAcquirer modelAcquirer;
+
+    @Inject
+    public ModelPathHelperImpl(Secrets secrets) {
+        this.secrets = secrets;
+        this.modelAcquirer = defaultModelAcquirer;
+    }
+
+    // For test purposes
+    ModelPathHelperImpl(Secrets secrets, ModelAcquirer modelAcquirer) {
+        this.secrets = secrets;
+        this.modelAcquirer = modelAcquirer;
+    }
+
+    private ModelAcquirer defaultModelAcquirer =
+         (urlReference, downloadOptions) -> {
+            UrlDownloader urlDownloader = new UrlDownloader();
+            File file = urlDownloader.waitFor(
+                    urlReference,
+                    downloadOptions,
+                    MODEL_DOWNLOAD_TIMEOUT
+            );
+
+            return Paths.get(file.getAbsolutePath());
+        };
+
+
+    public String getModelPathResolvingIfNecessary(ModelReference modelReference) {
+        if (isModelDownloadRequired(modelReference)) {
+            return resolveModelAndReturnPath(modelReference);
+        }
+
+        return modelReference.toString();
```

Comment:
To string will return the serialized version for the binary blob (config server -> config proxy communication). What we want is `ModelReference::value` or `ModelReference::path`.